### PR TITLE
Use new tool in mlir-aie

### PR DESCRIPTION
**Pull request by spaceotter**
Replaces the complex code to get the xclbin file with a call to a single tool.

### CODE REVIEW

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/AIETarget.cpp [hunk 1]
```diff
@@ -7,19 +7,23 @@
 #include "iree-amd-aie/Target/AIETarget.h"
 
 #include <filesystem>
```
Comment (MaheshRavishankar) [inline compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/AIETarget.cpp:9]: Are we use `filesystem` anywhere now. We can probably remove that?

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/AIETarget.cpp [hunk 2]
```diff
@@ -130,813 +135,118 @@ class AIETargetBackend final : public IREE::HAL::TargetBackend {
...
+  SmallVector<std::string> entryPointNames;
+  for (auto exportOp : variantOp.getExportOps()) {
+    entryPointNames.emplace_back(exportOp.getSymName());
   }
 
-  // Create a temporary work directory needed for subsequent commands.
-  FailureOr<std::string> workDir =
-      Artifact::createTemporaryDirectory(variantOp.getName());
-  if (failed(workDir)) {
+  if (entryPointNames.size() == 0) {
```
Comment (MaheshRavishankar) [inline compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/AIETarget.cpp:183]: Nit: You can just have `if (entryPointsNames.size()  == 1) { return ... "expected single entry point");}` for now.

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/AIETarget.cpp [hunk 3]
```diff
@@ -130,813 +135,118 @@ class AIETargetBackend final : public IREE::HAL::TargetBackend {
...
+    return moduleOp.emitOpError("Entry point missing");
+  } else if (entryPointNames.size() > 1) {
+    return moduleOp.emitOpError("Multiple entry points NYI");
+  }
+
+  SmallVector<StringRef> CmdArgs{aie2xclbin,
```
Comment (MaheshRavishankar) [inline compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/AIETarget.cpp:189]: Nit: `s/CmdArgs/cmdArgs`.

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/AIETarget.cpp [hunk 4]
```diff
@@ -130,813 +135,118 @@ class AIETargetBackend final : public IREE::HAL::TargetBackend {
...
-                              StringRef(dumpString.data(), dumpString.size()));
+    moduleOp.print(inputMlirOut->os(), OpPrintingFlags().useLocalScope());
+    inputMlirOut->keep();
   }
 
-  XclBinGeneratorKit toolkit(options.peanoInstallDir, options.vitisInstallDir,
-                             options.showInvokedCommands);
+  SmallString<128> aie2xclbin(options.mlirAieInstallDir);
+  llvm::sys::path::append(aie2xclbin, "bin", "aie2xclbin");
+  SmallString<128> XCLBinPath(workDir);
```
Comment (MaheshRavishankar) [inline compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/AIETarget.cpp:171]: Lets keep a consistent naming convention. All variable names start with lower case. So use `xclBinPath` and `ipuInstPath`, etc.

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/AIETarget.cpp [hunk 5]
```diff
@@ -130,813 +135,118 @@ class AIETargetBackend final : public IREE::HAL::TargetBackend {
...
-    OpBuilder::InsertionGuard g(executableBuilder);
-    executableBuilder.setInsertionPoint(moduleOp);
-    auto clonedModuleOp =
-        dyn_cast<ModuleOp>(executableBuilder.clone(*moduleOp.getOperation()));
-    if (failed(generateCoreElfFiles(moduleOp, objFile.value(), workDir.value(),
-                                    getOptions(), toolkit))) {
-      return failure();
+  std::vector<uint32_t> ipuInstrs;
+
+  std::ifstream instr_file(static_cast<std::string>(IPUInstPath));
```
Comment (MaheshRavishankar) [inline compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/AIETarget.cpp:234]: I am not sure why we are doing this? I would assume that we could just open it, read it as a file and just embed it without having to do any processing here.

**Review by MaheshRavishankar — CHANGES_REQUESTED** [review]
Very Nice! Great milestone! Just have a few comments.

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/AIETarget.cpp (hunk 5 above)
Comment (spaceotter) [reply compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/AIETarget.cpp:234]: The function I call in the tool writes hexadecimal.

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/AIETarget.cpp (hunk 4 above)
Comment (MaheshRavishankar) [reply compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/AIETarget.cpp:171]: I think MLIR does something differently. Thats what we tend to follow. Here too there is a mix with some local variables capitalized and some not. I think not capitalized is the convention.

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/AIETarget.cpp (hunk 5 above)
Comment (MaheshRavishankar) [reply compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/AIETarget.cpp:234]: I thought even the binary IREE writes to vmfb in hex. @nirvedhmeshram can you comment on this.

Comment (nirvedhmeshram) [reply compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/AIETarget.cpp:234]: I think the file is a string of hex values, and we are converting it to a vector of uint32_t for the vmfb here. So it seems required, unless the tool can allow to keep the vector in memory and we never go to disk.

**Review by MaheshRavishankar — APPROVED** [review]

**Pull request merged**